wandb/integration/ultralytics/callback.py

```diff
@@ -0,0 +1,367 @@
+                    epoch=trainer.epoch,
+                )
+            elif isinstance(trainer, DetectionTrainer):
+                self.train_validation_table = plot_validation_results(
+                    dataloader=dataloader,
+                    class_label_map=class_label_map,
+                    predictor=self.predictor,
+                    table=self.train_validation_table,
+                    max_validation_batches=self.max_validation_batches,
+                    epoch=trainer.epoch,
+                )
+            elif isinstance(trainer, ClassificationTrainer):
+                self.train_validation_table = plot_classification_validation_results(
+                    dataloader=dataloader,
+                    predictor=self.predictor,
+                    table=self.train_validation_table,
+                    max_validation_batches=self.max_validation_batches,
+                    epoch=trainer.epoch,
+                )
+        if self.enable_model_checkpointing:
+            self._save_model(trainer)
+        trainer.model.to(self.device)
+
+    def on_train_end(self, trainer: DetectionTrainer):
+        if isinstance(trainer, DetectionTrainer) or isinstance(
+            trainer, ClassificationTrainer
+        ):
+            wandb.log({"Train-Validation-Table": self.train_validation_table})
+
+    def on_val_end(self, trainer: DetectionValidator):
+        validator = trainer
+        dataloader = validator.dataloader
+        class_label_map = validator.names
+        with torch.no_grad():
+            self.predictor.setup_model(model=self.model, verbose=False)
+            if isinstance(trainer, PoseValidator):
+                self.validation_table = plot_pose_validation_results(
+                    dataloader=dataloader,
+                    class_label_map=class_label_map,
+                    predictor=self.predictor,
+                    visualize_skeleton=self.visualize_skeleton,
+                    table=self.validation_table,
+                    max_validation_batches=self.max_validation_batches,
+                )
+            elif isinstance(trainer, DetectionValidator):
+                self.validation_table = plot_validation_results(
+                    dataloader=dataloader,
+                    class_label_map=class_label_map,
+                    predictor=self.predictor,
+                    table=self.validation_table,
+                    max_validation_batches=self.max_validation_batches,
+                )
+            elif isinstance(trainer, ClassificationValidator):
+                self.validation_table = plot_classification_validation_results(
+                    dataloader=dataloader,
+                    predictor=self.predictor,
+                    table=self.validation_table,
+                    max_validation_batches=self.max_validation_batches,
+                )
+        wandb.log({"Validation-Table": self.validation_table})
```

Comment:
What get logged here if task is `segmentation` ? I'm guessing an empty table gets logged. Perhaps add a check to see if table has at least one row in data ?